source/common/network/address_impl.h

```diff
@@ -245,6 +250,32 @@ class PipeInstance : public InstanceBase {
   PipeHelper pipe_;
 };
 
+class EnvoyInternalInstance : public InstanceBase {
+public:
+  /**
+   * Construct from a string name.
+   */
+  explicit EnvoyInternalInstance(const std::string& envoy_listener_name,
+                                 const SocketInterface* sock_interface = nullptr);
+
+  // Network::Address::Instance
+  bool operator==(const Instance& rhs) const override;
+  const Ip* ip() const override { return nullptr; }
+  const Pipe* pipe() const override { return nullptr; }
+  const EnvoyInternalAddress* envoyInternalAddress() const override { return &internal_address_; }
+  const sockaddr* sockAddr() const override { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
```

Comment:
I'm mildly concerned that a panic() isn't the right call here.  Can you take a peek at call sites and see what we should do here to make sure these can never be called for the new address type?

We should probably also update the comments in the include directory to make it clear when it's safe to call and when it's not.

Reply:
With NOT_IMPLEMENTED_GCOVR_EXCL_LINE it means
If the later version of control plane send the config containing `EnvoyInternalAddress`, 
this particular version with this PR but no further of envoy might crash.

Reply:
What I can do is to find the potential usage of
 this new type of address. 
Listener and Endpoint as a immediate candidate, making sure these are either 
invalidate config per config update, mostly in master thread.
or fail the connection when this address is used. 

What else can I do?

Reply:
I think if we are making implementation of parts of the interface optional (panic if called), we either need to have TODOs to implement later or TODOs to check all call sites to make them internal-address compatible.  Having bits of an interface which may work for some address and may not work for others, and no checks in the code base makes me uncomfortable.  Again cc @mattklein123 in case he has other ideas of how to aim the decaped TCP stream back into a new envoy filter chain without a half-working address class.

Reply:
Updated to return nullptr and I mentally go through all the `sockAddr()` usage in the existing code base: all of them accepts nullptr usage without crash.
Luckily these usages are inline followed by syscall (connect/bind/send/recv) so I could inject syscall error code as if the following syscalls fail.
Now I believe these errors are now non-fatal.